### PR TITLE
Issue #15713: updated JavadocParagraph's input files based on issue description

### LINKED ISSUE #15713: update JavadocParagraph's input files
from https://github.com/checkstyle/checkstyle/pull/15712#pullrequestreview-2333617780

Two items:
- move all `// violation .....` out of javadoc
- resolve https://github.com/checkstyle/checkstyle/pull/15712#discussion_r1778036890 by split some code to new Input file 
- `//  '\<p\> tag should be placed immediately before the first word'` not clear why escaping is required.
- remove `/^ WARN/` text as they can affect validation

### PR DESCRIPTION
**Pull request by Zopsss**
fixes #15713

### CODE REVIEW

File: config/checkstyle-resources-suppressions.xml [hunk 1]
```diff
@@ -256,10 +256,6 @@
     files="[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]javadocmethod[\\/]InputJavadocMethodPublicOnly1\.java"/>
   <suppress checks="FileLength"
     files="[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]javadocmethod[\\/]InputJavadocMethodProtectedScopeJavadoc\.java"/>
-  <suppress checks="FileLength"
-    files="[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]javadocparagraph[\\/]InputJavadocParagraphIncorrect\.java"/>
-  <suppress checks="FileLength"
-    files="[\\/]test[\\/]resources[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]javadocparagraph[\\/]InputJavadocParagraphIncorrect2\.java"/>
```
Comment (Zopsss) [inline config/checkstyle-resources-suppressions.xml:262]: separated InputJavadocParagraphIncorrect2 in 2 files a( InputJavadocParagraphIncorrect2 & InputJavadocParagraphIncorrect3 )

InputJavadocParagraphIncorrect became small so didn't needed to split it

**Comment by Zopsss** [comment]
https://github.com/checkstyle/checkstyle/actions/runs/11085472433/job/30801929655?pr=15717#step:7:23

pitest is indicating JavadocTokenTypes.EOF is safe to delete. @romani

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect3.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        //  '<p> tag should be placed immediately before the first word'
+        //  '<p> tag should be preceded with an empty line.'
+        boolean emulated() {return false;}
+
+        // violation 3 lines below 'Empty line should be followed by <p> tag on the next line.'
+        /**
+         * Double newline.
+         *
+         *
+         * Some Javadoc. //DOUBLE WARN AT TWO PREVIOUS LINES
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect3.java:47]: Please remove trailing comment from javadoc ,

**Review by romani — CHANGES_REQUESTED** [review]
We need green ci

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java [hunk 3]
```diff
@@ -3,7 +3,7 @@
 /**
  * Some Javadoc.
  *
- * <p>/^ WARN/ Some Javadoc.
+ * <p>Some Javadoc.
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:6]: Please change `Some Javadoc.`  to ` some paragraph`.
And summary of javadoc to "Some summary.`.
To make it clear what is what in javadoc content.

To hint user that it is paragraph text.
Please apply to all.

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by romani** [comment]
```
+  <mutation unstable="false">
+    <sourceFile>JavadocParagraphCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocParagraphCheck</mutatedClass>
+    <mutatedMethod>isImmediatelyFollowedByText</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator</mutator>
+    <description>removed call to com/puppycrawl/tools/checkstyle/api/DetailNode::getType</description>
+    <lineContent>return nextSibling.getType() == JavadocTokenTypes.EOF</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>JavadocParagraphCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocParagraphCheck</mutatedClass>
+    <mutatedMethod>isImmediatelyFollowedByText</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.RemoveConditionalMutator_EQUAL_IF</mutator>
+    <description>removed conditional - replaced equality check with true</description>
+    <lineContent>return nextSibling.getType() == JavadocTokenTypes.EOF</lineContent>
+  </mutation>
```

this means we lost test case (some code in Input) that kept EOF meanigful.

**Comment by Zopsss** [comment]
> please recheck that we still have unrealistic javadoc `...Javadoc.<p>*/`

we have this suppression:

https://github.com/checkstyle/checkstyle/blob/a8db72d7330b43e8a9ad08acdcd8f71fd1fe8e17/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java#L47-L54

**Comment by romani** [comment]
I am confused, we just changed input files,   and it means we just lost test case that covered EOF.
Please revert charges on your local by files to catch what file covered EOF. And later on revert charges in file to find exact javadoc that use to cover EOF.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect3.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        //  '<p> tag should be placed immediately before the first word'
+        //  '<p> tag should be preceded with an empty line.'
+        // violation 4 lines below '<p> tag should be placed immediately before the first word'
+        /**
+         *   Some Javadoc.<p>
+         *
+         *  <p>  Some Javadoc.
+         *
+         * @see <a href="http://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideModules">
+         *     Documentation about <p> GWT emulated source</a>
```
[the remark this answers is not in the document]
Comment (Zopsss) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect3.java:35]: done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect3.java (hunk 2 above)
Comment (Zopsss) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect3.java]: done

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java (hunk 3 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputFormattedIncorrectJavadocParagraph.java:6]: done

**Comment by Zopsss** [comment]
> I am confused, we just changed input files, and it means we just lost test case that covered EOF. Please revert charges on your local by files to catch what file covered EOF. And later on revert charges in file to find exact javadoc that use to cover EOF.

I added this test case again:

https://github.com/checkstyle/checkstyle/blob/5b9c6b76736d9b78bae2735cc4846b00be414f38/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java#L99-L104

without `EOF` test started failing, I guess we will have green ci this time.

**Review by romani — APPROVED** [review]
Awesome step forward.

@rdiachenko , I am merging this by single review, it is just inputs clean up

**Pull request merged**